aten/src/ATen/CMakeLists.txt

```diff
@@ -112,7 +112,7 @@ file(GLOB native_sparse_cpp "native/sparse/*.cpp")
 file(GLOB native_quantized_cpp
             "native/quantized/*.cpp"
             "native/quantized/cpu/*.cpp")
-file(GLOB native_nested_cpp "native/nested/*.cpp")
+file(GLOB native_nested_cpp "native/nested/*.cpp" "native/nested/flash_attn/*.cpp")
```

Comment:
I suspect this is causing build issues, because some of the .cpp files in that folder pull in CUDA headers (e.g. `#include <ATen/cuda/CUDAContext.h>` in fmha_api.cpp) and we separate our CPU and CUDA builds. One way around this is having `flash_scaled_dot_product_self_attention` only registered under the CUDA and NestedTensorCUDA flags and generally hove the whole thing under `native/nested/cuda/flash_attn`.

Reply:
updated